File: Java/Tema4_Ejercicio9.java
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 9
 * ===============================================
 * Escriba un programa que use continue para saltar números pares 
 * e imprimir solo los números impares del 1 al 20.
 */

public class Tema4_Ejercicio9 {
    public static void main(String[] args) {
        System.out.println("=== NÚMEROS IMPARES DEL 1 AL 20 ===");
        
        for (int i = 1; i <= 20; i++) {
            if (i % 2 == 0) {
                continue;  // Salta los números pares
            }
            System.out.println("Número: " + i);
        }
        
        System.out.println("\n=== NÚMEROS PARES DEL 1 AL 20 ===");
        
        for (int i = 1; i <= 20; i++) {
            if (i % 2 != 0) {
                continue;  // Salta los números impares
            }
            System.out.println("Número: " + i);
        }
    }
}

/*
 * SALIDA ESPERADA:
 * === NÚMEROS IMPARES DEL 1 AL 20 ===
 * Número: 1
 * Número: 3
 * Número: 5
 * ... (hasta 19)
 * 
 * === NÚMEROS PARES DEL 1 AL 20 ===
 * Número: 2
 * Número: 4
 * Número: 6
 * ... (hasta 20)
 */

File: JS/Tema4_Ejercicio9.js
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 9
 * ==============================================
 * Filtrar pares e impares con continue.
 * 
 * EXPLICACIÓN:
 * - Continue para saltar a la siguiente iteración
 * - Filtrado de números según una condición
 * - Separación de pares e impares
 */

console.log("=== USO DE CONTINUE ===");

const numeros = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
const pares = [];
const impares = [];

console.log("Números originales:", numeros.join(", "));

// Usando continue para filtrar
for (let num of numeros) {
    if (num % 2 !== 0) {
        impares.push(num);
        continue;  // Salta sin ejecutar el resto
    }
    pares.push(num);
}

console.log("\nNúmeros PARES:", pares.join(", "));
console.log("Números IMPARES:", impares.join(", "));

// Otra forma: saltando números específicos
console.log("\n=== BUCLE CON CONTINUE ===");
console.log("Números pares del 1 al 10 (saltando 6):");
for (let i = 1; i <= 10; i++) {
    if (i % 2 !== 0) continue;  // Salta impares
    if (i === 6) continue;       // Salta el 6
    console.log(i);
}

/*
 * SALIDA ESPERADA:
 * === USO DE CONTINUE ===
 * Números originales: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15
 * 
 * Números PARES: 2, 4, 6, 8, 10, 12, 14
 * Números IMPARES: 1, 3, 5, 7, 9, 11, 13, 15
 * 
 * === BUCLE CON CONTINUE ===
 * Números pares del 1 al 10 (saltando 6):
 * 2
 * 4
 * 8
 * 10
 */
